x-pack/osquerybeat/internal/osqdcli/client.go

```diff
@@ -259,7 +273,7 @@ func (c *Client) resolveResult(ctx context.Context, sql string, hits []map[strin
 
 func (c *Client) queryColumnTypes(ctx context.Context, sql string) (map[string]string, error) {
 	colTypes, ok := c.cache.Get(sql)
-	if ok {
+	if ok && c.log != nil {
 		c.log.Debugf("using cached column types for query: %s", sql)
 	}
```

Comment:
Could this client also be used to create the views?  I would assume so.

Reply:
we could but I am not sure we need the added complexity given our use case in the extension, this also  brings elastic-agent-libs which might add to the extension binary size) is there any particular feature of it that you find interesting?